client/directclient.go

```diff
@@ -63,15 +65,35 @@ func (c *ProxyDirectClient) quorumResponse(reqs ...*http.Request) (http.Header,
 	}
 	quorum := int(math.Ceil(float64(len(reqs)) / 2.0))
 	responseClasses := []int{0, 0, 0, 0, 0, 0}
+	responseCount := 0
+	var chosenResponse *QuorumResponseEntry
 	for response := range responses {
+		responseCount++
 		class := response.StatusCode / 100
 		if class <= 5 {
 			responseClasses[class]++
 			if responseClasses[class] >= quorum {
-				return response.Headers, response.StatusCode
+				chosenResponse = response
+				break
+			}
+		}
+	}
+	// give any pending requests *some* chance to finish
+	if responseCount < len(reqs) {
+		timeout := time.After(PostQuorumTimeoutMs * time.Millisecond)
+		select {
+		case <-responses:
+			responseCount++
+			if responseCount == len(reqs) {
+				break
 			}
+		case <-timeout:
+			break
```

Comment:
This break and the other one aren't really doing anything.  It breaks out of the "select", but go doesn't fallthrough cases anyway.